_core/smart/principals.md

```diff
@@ -14,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
+"owned" by objects of the principal type, meaning that any object of
+the principal type may own an asset. For the case of public-key hash
+and multi-signature Stacks addresses, a given principal can operate on
+their assets by issuing a signed transaction on the blockchain. _Smart
+contracts_ may also be principals (reprepresented by the smart
+contract's identifier), however, there is no private key associated
+with the smart contract, and it cannot broadcast a signed transaction
+on the blockchain.
+
+A Clarity contract can use a globally defined `tx-sender` variable to
+obtain the current principal. The following example defines a transaction
+type that transfers `amount` uSTX from the sender to a recipient if amount
+is a multiple of 10, otherwise returning a 400 error code.
+
+```scheme
+(define-public (transfer-to-recipient! (recipient principal) (amount uint))
+  (if (is-eq (mod amount 10) 0)
+      (stx-transfer? amount tx-sender recipient)
+      (err u400)))
 ```
 
-The principal's signature is not checked by the smart contract, but by the virtual machine.
-
-
 ## Smart contracts as principals
 
-Smart contracts themselves are principals and are represented by the smart contract's identifier. You create the identifier when you launch the contract, for example, the contract identifier here is `hanomine`.
+Smart contracts themselves are principals and are represented by the
+smart contract's identifier -- which is the publishing address of the
+contract _and_ the contract's name, e.g.:
 
-```bash
-clarity-cli launch hanomine /data/hano.clar /data/db
+```scheme
+'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR.contract-name
 ```
 
-A smart contract may use the special variable `contract-name` to refer to its own principal.
+For convenience, smart contracts may write a contract's identifier in the
+form `.contract-name`, which is expanded by the Clarity interpreter into
+a fully-qualified contract identifier that corresponds to the same
+publishing address as the contract it appears in.
 
-To allow smart contracts to operate on assets it owns, smart contracts may use the special `(as-contract expr)` function. This function executes the expression (passed as an argument) with the `tx-sender` set to the contract's principal, rather than the current sender. The `as-contract` function returns the value of the provided expression.
+In order for a smart contract to operate on assets it owns, smart contracts
+may use the special `(as-contract ...)` function. This function
+executes the expression (passed as an argument) with the `tx-sender`
+set to the contract's principal, rather than the current sender. The
+`as-contract` function returns the value of the provided expression.
 
-For example, a smart contract that implements something like a "token faucet" could be implemented as so:
+For example, a smart contract that implements something like a "token
+faucet" could be implemented as so:
 
-```cl
+```scheme
```

Comment:
does scheme work better? I think "cl" is used for most coding sections in the docs

Reply:
neither are perfect, but I've mostly found scheme to work a little better than cl